gax/retry_loop_test.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gax/retry_loop.h"
+#include "google/longrunning/operations.pb.h"
+#include "googletest/include/gtest/gtest.h"
+#include "gax/backoff_policy.h"
+#include "gax/call_context.h"
+#include "gax/retry_policy.h"
+#include <chrono>
+
+namespace {
+using namespace ::google;
+
+std::unique_ptr<gax::RetryPolicy> ErrCountRetryFactory(int n) {
+  return std::unique_ptr<gax::LimitedErrorCountRetryPolicy>(
+      new gax::LimitedErrorCountRetryPolicy(n));
+}
+
+TEST(RetryLoop, Basic) {
```

Comment:
This is a very big test, called `Basic`. Is it really basic?  Can it be split?

Reply:
It's only a big test because the class definition is in it. I moved the class outside the test, and now it looks much more like a Basic test.